File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/LastPositionOfInvocation.java
package org.cqframework.cql.cql2elm.model.invocation;

import java.util.Arrays;
import java.util.List;
import org.hl7.elm.r1.Expression;
import org.hl7.elm.r1.LastPositionOf;

public class LastPositionOfInvocation extends OperatorExpressionInvocation<LastPositionOf> {
    public LastPositionOfInvocation(LastPositionOf expression) {
        super(expression);
    }

    @Override
    public List<Expression> getOperands() {
        return Arrays.asList(expression.getPattern(), expression.getString());
    }

    @Override
    public void setOperands(List<Expression> operands) {
        require(operands != null && operands.size() == 2, "LastPositionOf operator requires two operands.");
        expression.setPattern(operands.get(0));
        expression.setString(operands.get(1));
    }
}

File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/LastPositionOfInvocation.kt
package org.cqframework.cql.cql2elm.model.invocation

import org.hl7.elm.r1.Expression
import org.hl7.elm.r1.LastPositionOf

class LastPositionOfInvocation(expression: LastPositionOf) :
    OperatorExpressionInvocation<LastPositionOf>(expression) {
    override var operands: List<Expression>
        get() = listOf(expression.pattern, expression.string)
        set(operands) {
            require(operands.size == 2) { "LastPositionOf operator requires two operands." }
            expression.pattern = operands[0]
            expression.string = operands[1]
        }
}
